events/code_review_events/workflow.py

```diff
@@ -553,15 +534,25 @@ def run(self):
         if self.webserver:
             self.webserver.start()
 
+        loop = asyncio.get_event_loop()
+
         if consumers:
             # Run all tasks concurrently
-            run_tasks(consumers)
+            try:
```

Comment:
This whole block could be implemented in libmozevent, straight into `run_tasks`. The call to `restore_redis_messages` should simply be parametrizable (to be backward compatible)